### PR TITLE
New Autofill pixels for KPIs

### PR DESCRIPTION
**Pull request by amddg44**
Task/Issue URL: https://app.asana.com/0/0/1206508535741136/f
Tech Design URL:
CC:

**Description**:
Adds new pixels as part of unified autofill KPIs across platforms

**Steps to test this PR**:

**NOTE** - if you have an existing build with 10 or more passwords, feel free to start with Step 9 & 10 first 🙂 
1. Starting with a new install of the app, visit https://fill.dev/form/registration-username and save a new login using a generated password (triggering a fill event)
2. Confirm this pixel fires: `m.autofill.logins.stacked ["count_bucket": "none"]`
3. Perform a search in the browser
4. Confirm this pixel fires: `m.autofill.activeuser` (defined as a user that has performed both an autofill fill action & a search today)
5. Re-launch the app and confirm this pixel fires: `m.autofill.onboardeduser`  (defined as user who has stored at least 1 password within the first week)
6. Perform another search and confirm none of the above pixels fire again
7. Visit to https://fill.dev/form/login-simple and fill the previously saved login and again confirm none of the above pixels fire again
8. Delete and re-install the app
9. Save 10+ logins and then perform a search
10. Confirm this pixel fires: `m.autofill.enableduser` (defined as a user who is a search DAU and has at least 10 passwords saved)

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo/AutofillPixelReporter.swift [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+                DailyPixel.fire(pixel: pixel, withAdditionalParameters: [PixelParameters.countBucket: bucket])
+            default:
+                DailyPixel.fire(pixel: pixel)
+            }
+        }
+    }
+
+    private func shouldFireActiveUserPixel() -> Bool {
+        let today = Date()
+        if today.isSameDay(autofillSearchDauDate) && today.isSameDay(autofillFillDate) {
```
Comment (aataraxiaa) [inline DuckDuckGo/AutofillPixelReporter.swift:114]: If you wanted you could change this to:

```suggestion
        if Date().isSameDay(autofillSearchDauDate) && today.isSameDay(autofillFillDate) {
```

…just to be consistent with how you use `Date()` in the other functions above and below

File: DuckDuckGoTests/AutofillPixelReporterTests.swift [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+import XCTest
+@testable import DuckDuckGo
+@testable import Core
+@testable import BrowserServicesKit
+
+final class AutofillPixelReporterTests: XCTestCase {
+
+    var statisticsStorage: MockStatisticsStore!
+    private let vault = (try? MockSecureVaultFactory.makeVault(reporter: nil))!
+    var autofillPixelReporter: AutofillPixelReporter!
```
Comment (aataraxiaa) [inline DuckDuckGoTests/AutofillPixelReporterTests.swift]: Could make these private for consistency if you preferred:

```suggestion
    private var statisticsStorage: MockStatisticsStore!
    private let vault = (try? MockSecureVaultFactory.makeVault(reporter: nil))!
    private  var autofillPixelReporter: AutofillPixelReporter!
```

**Review by aataraxiaa — APPROVED** [review]
Tested and works great. All LGTM, just left two super minor suggestions.

File: DuckDuckGo/AutofillPixelReporter.swift (hunk 1 above)
Comment (amddg44) [reply DuckDuckGo/AutofillPixelReporter.swift:114]: ah, that was actually intentional as I'm referencing `today` twice in that if condition 🙂

**Pull request merged**